src/app/view/timestamp-editor/timestamp-editor.component.ts

```diff
@@ -144,8 +144,16 @@ export class TimestampEditorComponent implements OnInit {
   }
 
   showVesselPosition(): boolean {
-    if (!this.globals.config.enableVesselPositions) return false;
-    return this.timestampTypeSelected?.value?.isVesselPositionNeeded ?? false;
+    let validators = null;
+    const timestampTypeSelected = this.timestampTypeSelected?.value;
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED) {
+      validators = [Validators.required];
+    }
+    this.timestampFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+    this.timestampFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+    this.timestampFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+    this.timestampFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+    return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```

Comment:
Getter should not have side-effects.

Reply:
This is how we have been defining our "show" methods on the component, for example the method `showTerminalOption()`

Reply:
"Two wrongs do not make a right".